app/domain/authentication/authn_k8s/k8s_object_lookup.rb

```diff
@@ -104,11 +104,16 @@ def pod_by_name(podname, namespace)
         k8s_client_for_method("get_pod").get_pod(podname, namespace)
       end
 
-      # Locates the Namespace with a given name.
+      # Returns the labels hash for a Namespace with a given name.
       #
       # @return nil if no such Namespace exists.
-      def namespace_by_name(namespace)
-        k8s_client_for_method("get_namespace").get_namespace(namespace)
+      def namespace_labels_hash(namespace)
+        namespace_object = k8s_client_for_method("get_namespace").get_namespace(namespace)
+        if !namespace_object
+          return nil
+        end
+
+        namespace_object.metadata.labels.to_h
```

Comment:
nit: I think this might currently raise a lint error. I think this would have the same outcomes:
```rb
return namespace_object.metadata.labels.to_h unless namespace_object.nil?
```